### PR TITLE
Promote first GGUF text architecture cohort

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

Promote five previously `immediately-implementable` GGUF architecture routes with architecture-owned metadata, tensor, graph, and fail-closed capability contracts:

- `granite`: exact dense/MoE union, Granite scaling, optional biases/shared experts, fused/split QKV transforms, tied output, and validated LongRoPE/no-RoPE paths. Packed import remains supported only where the transforms are proven; unmapped YaRN metadata is rejected.
- `jina-bert-v3`: dedicated dense RoPE encoder with fused/split QKV closure, post-norm GELU blocks, pooling, singleton token-type transform, and mandatory encoder-only task/cache guards. Unreachable MoE and packed preservation are rejected.
- `maincoder`: dedicated post-RoPE learned Q/K RMSNorm decoder with tied output and exact dynamic/static KV semantics. Packed import is rejected.
- `plamo`: shared-pre-norm parallel attention/SwiGLU decoder, cyclic 5-to-40-head KV expansion/cache, and inverse converter Q/output shuffles. Packed preservation, static cache, and external weight reuse are rejected.
- `smallthinker`: unnormalized-input routing, ReGLU experts, softmax/sigmoid gating normalization, and exact SWA/NoPE/local-RoPE schedules. Non-integral gating enums and unproven routing scales fail closed; packed import and static cache are rejected.

All five routes keep downstream runtime packaging explicitly deferred pending representative real-weight parity. No production routing pins, defaults, or rejects Hugging Face revisions by `model_id`; `revision=None` and arbitrary explicit revisions are unchanged. Immutable Maincoder/Jina revisions are test-only. PLaMo/SmallThinker use explicit model-coverage exemptions because their exact semantics are restored only by GGUF metadata postprocessing, not a native HF config alias.

The generated GGUF census/docs now remove these routes from `immediately-implementable` and record their exact import, quantization, and runtime dispositions.

## Review corrections

- Exclude `encoder_fused_qkv` from unrelated route fingerprints while retaining it for Jina-v3; regression coverage proves the llama, qwen2, lfm2, and qwen35moe evidence routes are invariant.
- Add immutable test-only L2 IDs/revisions for Maincoder and Jina-v3; document explicit GGUF-only exemptions for PLaMo and SmallThinker.
- Add Jina-v3 to the encoder-only guard so static cache and incompatible task overrides fail before task dispatch.
- Reject Granite YaRN until every serialized parameter, including `attn_factor`, has an exact mapping.
- Require SmallThinker gating enums to be finite integers and reject any nonzero `expert_weights_scale` absent from the pinned loader contract.

## Validation

- Rebased onto `origin/main` `20a29ed2a935f01ec0cf2fc82939e50bc71dcdea`.
- Five-finding regression selection: **96 passed, 5 skipped**.
- Focused cohort suite: **575 passed, 33 skipped**.
- Exact model coverage: **786 passed, 257 justified skips**.
- Pinned Maincoder/Jina L2 config and full-graph validation: **6 passed**.
- Broad serial non-integration suite: **8,703 passed, 64 skipped, 12 deselected**.
- Generated GGUF docs/registry suite: **1,146 passed, 1 skipped**.
- Full `lintrunner` formatting/lint and `git diff --check` pass.
- Configured mypy was rerun; it reports the repository's existing type-error baseline, with no diagnostic on a changed line. The new route modules retain their prior targeted type-check evidence.
- Independent review of the complete correction diff found no significant issues.

## Artifact budget and cleanup

Recorded bounded artifact/source reads total at most **34,042,487 bytes** plus small metadata responses, far below the **16 GiB** limit. Only two 16 MiB GGUF ranges were read (Maincoder and JinaBERT-v3); Granite, PLaMo, and SmallThinker downloaded no model payloads. The correction pass downloaded config-only L2 metadata, not model weights. Sparse headers, pinned source scratch files, transient outputs, caches, and temporary packages were deleted. No `.gguf`, `.onnx`, `.safetensors`, or model `.bin` payload remains in the worktree.

## Deferred evidence / waivers

- Representative real-weight full-logit prefill/cached-decode parity and deterministic generation remain deferred for all five routes; suitable immutable artifacts either exceed the per-route budget or were not required to prove the narrowed import contract.
- ORT GenAI/runtime packaging, CUDA/DML, L4/L5 goldens, and deployment claims remain deferred.
- JinaBERT-v3, Maincoder, PLaMo, and SmallThinker native packed quantization remain rejected. Granite packed import is covered by synthetic packed graph/value evidence, not a real-weight runtime parity claim.
- PLaMo’s smallest immutable public GGUF is 4,910,733,280 bytes, above the 3 GiB route ceiling, so no payload was downloaded.

### CODE REVIEW

File: tests/gguf_plamo_parity_test.py [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+            key = torch.cat((layer_past[0], key), dim=2)
+            value = torch.cat((layer_past[1], value), dim=2)
+        presents.append((key, value))
+
+        scores = torch.matmul(query.transpose(1, 2), key.transpose(-1, -2))
+        scores = scores * config.head_dim**-0.5
+        key_positions = torch.arange(key.shape[2], dtype=torch.int64)
+        causal = key_positions[None, :] <= position_ids.reshape(-1, 1)
+        valid = attention_mask.bool().reshape(batch, 1, 1, -1)
+        scores = scores.masked_fill(~(causal[None, None, :, :] & valid), float("-inf"))
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply tests/gguf_plamo_parity_test.py]: Fixed in `4672679a`: the independent PLaMO oracle now constructs causal masks as `(batch, seq, total_k)` and broadcasts them as `(batch, 1, seq, total_k)`. A two-row full-logit ORT parity regression covers the corrected batch semantics.

File: src/mobius/integrations/gguf/_config_mapping.py [hunk 2]
```diff
@@ -1483,6 +1487,131 @@ def _moe_postprocess(
...
 
+def _smallthinker_postprocess(
+    config: ArchitectureConfig,
+    metadata: dict[str, Any],
+    model: Any,
+) -> ArchitectureConfig:
+    """Restore the pinned llama.cpp SmallThinker routing and layer schedule."""
+    arch = "smallthinker"
+    gating = int(metadata[f"{arch}.expert_gating_func"])
+    if gating not in (1, 2):
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/gguf/_config_mapping.py]: Fixed in `2bf23558`: SmallThinker now converts the enum to a float only for validation, requires it to be finite and integral before calling `int`, and rejects `1.5` with a focused regression.
